pkg/version/version.go

```diff
@@ -0,0 +1,47 @@
+package version
+
+import (
+	"k8s.io/apimachinery/pkg/version"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	// commitFromGit is a constant representing the source version that
+	// generated this build. It should be set during build via -ldflags.
+	commitFromGit string
+	// versionFromGit is a constant representing the version tag that
+	// generated this build. It should be set during build via -ldflags.
+	versionFromGit string
+	// major version
+	majorFromGit string
+	// minor version
+	minorFromGit string
+	// build date in ISO8601 format, output of $(date -u +'%Y-%m-%dT%H:%M:%SZ')
+	buildDate string
+)
+
+// Get returns the overall codebase version. It's for detecting
+// what code a binary was built from.
+func Get() version.Info {
+	return version.Info{
+		Major:      majorFromGit,
+		Minor:      minorFromGit,
+		GitCommit:  commitFromGit,
+		GitVersion: versionFromGit,
+		BuildDate:  buildDate,
+	}
+}
+
+func init() {
+	buildInfo := prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "openshift_cert_manager_operator_build_info",
+			Help: "A metric with a constant '1' value labeled by major, minor, git commit & git version from which OpenShift Cert Manager Operator was built.",
+		},
+		[]string{"major", "minor", "gitCommit", "gitVersion"},
+	)
+	buildInfo.WithLabelValues(majorFromGit, minorFromGit, commitFromGit, versionFromGit).Set(1)
+
+	prometheus.MustRegister(buildInfo)
+}
```

Comment:
this might be getting reported here?

https://github.com/openshift/library-go/blob/28a7a4065a9e0070ef60c661f37903fa875ac332/pkg/controller/controllercmd/builder.go#L227-L241

Reply:
i think i took this over from somewhere, but you know what it just occured to me: we really should dependency-inject the registry and omit the global init.

Reply:
and let me recheck if we hit the above library-go function

Reply:
ok, nice, yes, the builder auto-injects the build info already :+1: 
```
$ curl -s -k -H "Authorization: Bearer $(kubectl -n openshift-monitoring get secret $(kubectl -n openshift-monitoring get sa prometheus-k8s -o jsonpath='{.secrets[0].name}') -o jsonpath='{.data.token}' | base64 -d)" https://localhost:8443/metrics | grep build_info
# HELP openshift_cert_manager_operator_build_info [ALPHA] A metric with a constant '1' value labeled by major, minor, git version, git commit, git tree state, build date, Go version, and compiler from which openshift-cert-manager-operator was built, and platform on which it is running.
# TYPE openshift_cert_manager_operator_build_info gauge
openshift_cert_manager_operator_build_info{buildDate="2021-06-18T15:47:43Z",compiler="",gitCommit="49c5269",gitTreeState="",gitVersion="v0.0.0-unknown",goVersion="",major="",minor="",platform=""} 1
```